addons/web_editor/static/src/js/editor/odoo-editor/src/powerbox/Powerbox.js

```diff
@@ -248,39 +270,65 @@ export class Powerbox {
+     * existing category names and whose values are each of these categories'
+     * commands. Categories with no commands are removed.
      *
      * @private
      * @param {PowerboxCommand[]} commands
-     * @returns {{[key: groupName]: PowerboxCommand[]}>}
+     * @param {Array<{name: string, priority: number}} categories
+     * @returns {{Array<[string, PowerboxCommand[]]>}>}
      */
-    _groupCommands(commands) {
-        const groups = {};
-        for (const command of commands) {
-            groups[command.groupName] = [...(groups[command.groupName] || []), command];
+    _groupCommands(commands, categories) {
+        const groups = [];
+        for (const category of this._orderByPriority(categories)) {
+            const categoryCommands = commands.filter(command => command.category === category.name);
+            commands = commands.filter(command => command.category !== category.name);
+            groups.push([category.name, this._orderByPriority(categoryCommands)])
         }
-        return groups;
+        // If commands remain, it means they declared categories that didn't
+        // exist. Add these categories alphabetically at the end of the list.
+        const remainingCategories = [...new Set(commands.map(command => command.category))];
+        for (const categoryName of remainingCategories.sort((a, b) => a.localeCompare(b))) {
+            const categoryCommands = commands.filter(command => command.category === categoryName);
+            groups.push([categoryName, this._orderByPriority(categoryCommands)]);
+        }
+        return groups.filter(group => group[1].length);
     }
     /**
      * Take a list of commands and returns them reordered so that commands that
-     * belong to the same group are grouped together.
+     * belong to the same category are grouped together.
      *
      * @private
      * @param {PowerboxCommand[]} commands
+     * @param {Array<{name: string, priority: number}} categories
      * @returns {PowerboxCommand[]}
      */
-    _orderCommandsByGroup(commands) {
-        const groups = this._groupCommands(commands);
+    _orderCommandsByCategory(commands, categories) {
+        categories = this._orderByPriority(categories);
         const reorderedCommands = [];
-        for (const groupCommands of Object.values(groups)) {
-            reorderedCommands.push(...groupCommands);
+        for (const group of this._groupCommands(commands, categories)) {
+            reorderedCommands.push(...group[1]);
         }
         return reorderedCommands;
     }
+    /**
+     * Take an array of commands or categories and return a reordered copy of
+     * it, based on their respective priorities.
+     *
+     * @param {PowerboxCommand[] | Array<{name: string, priority: number}} commandsOrCategories
+     * @returns {PowerboxCommand[] | Array<{name: string, priority: number}}
+     */
+    _orderByPriority(commandsOrCategories) {
```

Comment:
Thanks, this is great ! Now the method doesn't really serve a purpose anymore. It just basically apply the sort on a copy so I would have removed the method altogether and sort it like this at call site:
```js
const order = (a, b) => b.priority - a.priority || a.name.localeCompare(b.name);
categories = [...categories].sort(order);
commands = this._getCurrentCommands([commands...].sort(order), categories);
```
Since the arrow notation creates a new function at runtime anyway, this is equivalent to the current code of the PR in terms of new function creations. What do you think ?